components/brave_federated_learning/brave_operational_profiling.cc

```diff
@@ -0,0 +1,205 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/components/brave_federated_learning/brave_operational_profiling.h"
+
+#include "base/json/json_writer.h"
+#include "base/strings/string_util.h"
+#include "base/time/time.h"
+#include "base/timer/timer.h"
+#include "brave/components/brave_federated_learning/brave_operational_profiling_features.h"
+#include "brave/components/brave_stats/browser/brave_stats_updater_util.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+namespace {
+
+static constexpr char federatedLearningUrl[] = "https://fl.brave.com/";
```

Comment:
Is this domain already set up with Cloudflare Spectrum?

Reply:
We recently had [some folks on Reddit complain](https://www.reddit.com/r/CryptoCurrency/comments/nxce6t/brave_browser_scam_a_fake_privacy_browser_sharing/) about data being sent to weirdly-named URLs. That Reddit thread is a mess, but I think we should try to make sure data is being sent to well-named URLs. We currently send some usage data already, right? Could we use that domain?

Reply:
Good call out @ShivanKaul, I _fully_ agree with this. At the very least we should have a way for the user to see what this endpoint is collecting, or heck why it's even named the way it is. If we need to keep the domain we should at least have a resource for the end users to see what is going on.

Reply:
FYI @fmarier filed https://github.com/brave/brave-browser/issues/16374. Perhaps metrics.brave.com?
Just a note that the wiki link being sent as part of the payload will help a lot, but the simpler and clearer we make place + purpose for data leaving the browser, the likelier it is we'll avoid PR nuisances.

Reply:
Based on the conversation in https://github.com/brave/security/issues/479, this is fine.